vars/abapEnvironmentPipelineStageIntegrationTests.groovy

```diff
@@ -31,29 +32,42 @@ void call(Map parameters = [:]) {
         .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
         .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
         .mixin(parameters, PARAMETER_KEYS)
-        .addIfEmpty('confirmDeletion', true)
+        .addIfEmpty('confirmDeletion', false)
         .addIfEmpty('debug', false)
         .addIfEmpty('testBuild', false)
+        .addIfEmpty('integrationTestOption', 'addOnDeployment')
         .use()
 
     if (config.testBuild) {
         echo "Stage 'Integration Tests' skipped as parameter 'testBuild' is active"
     } else {
         piperStageWrapper (script: script, stageName: stageName, stashContent: [], stageLocking: false) {
-            try {
-                abapEnvironmentCreateSystem(script: parameters.script, includeAddon: true)
-                cloudFoundryCreateServiceKey(script: parameters.script)
-                abapEnvironmentBuild(script: parameters.script, phase: 'GENERATION', downloadAllResultFiles: true, useFieldsOfAddonDescriptor: '[{"use":"Name","renameTo":"SWC"}]')
-            } catch (Exception e) {
-                echo "Deployment test of add-on product failed."
-                throw e
-            } finally {
-                if (config.confirmDeletion) {
-                    input message: "Deployment test has been executed. Once you proceed, the test system will be deleted."
+            if (config.integrationTestOption == 'systemProvisioning') {
+                try {
+                    abapEnvironmentCreateSystem(script: parameters.script, includeAddon: true)
+                    cloudFoundryCreateServiceKey(script: parameters.script)
+                    abapEnvironmentBuild(script: parameters.script, phase: 'GENERATION', downloadAllResultFiles: true, useFieldsOfAddonDescriptor: '[{"use":"Name","renameTo":"SWC"}]')
+                } catch (Exception e) {
+                    echo "Deployment test of add-on product failed."
+                    throw e
+                } finally {
+                    if (config.confirmDeletion) {
+                        input message: "Deployment test has been executed. Once you proceed, the test system will be deleted."
+                    }
```

Comment:
shouldn't this be moved to the end, so at least theoretically deploy can be used with this option in self hosted pipelines as well?